docs/modules/ROOT/pages/utilities.adoc

```diff
@@ -188,3 +188,49 @@ await instance.multicall([
     instance.contract.methods.bar().encodeABI()
 ]);
 ----
+
+=== StaticDelegateCall
+
+The `StaticDelegateCall` abstract contract provides emulation of static delegatecalls for implementing view-only functions that execute another contract's code with the calling contract's context. It can be useful in implementing dynamic library support at runtime.
+
+Example:
+[source,solidity]
+----
+// contracts/DynamicGadget.sol
```

Comment:
Although its purpose is to showcase how the static delegatecall works, I'd say is difficult to follow because the purpose of `Gadget` is unclear.

Do you think we can provide a cleaner example? Maybe a Proxy would work here but I understand if it's too complex

Reply:
In our case, we were instantiating a contract with a possibly user specified "dynamic library" at runtime (i.e. not inlined at compile time, nor externally linked at deploy time), that would act on contract storage but do implementation-specific things, while still conforming to an interface like `IGadget`. But we ran into the issue of not being able to write any view functions.

Our use case might eventually be solved at the language level, e.g. ethereum/solidity#8490, ethereum/solidity#2469, ethereum/solidity#8353, but there are other uses for a static delegate call, like emulating external logic, providing generic access to contract storage, and from my research it appears other people were running into this issue in the past: https://ethereum.stackexchange.com/questions/71037/anything-like-a-static-delegatecall-from-a-view-function https://ethereum.stackexchange.com/questions/82342/how-to-perform-delegate-call-inside-of-view-call-staticall . I'll try to think of a better example.

Reply:
> But we ran into the issue of not being able to write any view functions.

What do you mean with that? Can you provide an example? I think the example could be simplified to only using a `staticDelegateCall` regardless of the use case and then listing the use cases after the example.

Unless there's a relevant security consideration to account, simple would be better at first sight.

Reply:
> What do you mean with that? Can you provide an example? I think the example could be simplified to only using a staticDelegateCall regardless of the use case and then listing the use cases after the example.

It's simply that you can't write a `view` function that uses `delegatecall`, so there's no way to write a getter that uses a contract's storage with another contract's logic at runtime. (Externally linked libraries can, but they have to be linked at deploy time.)